src/components/NewTodoField/NewTodoField.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { memo, useEffect } from 'react';
+
+type Props = {
+  newTodoField?: React.RefObject<HTMLInputElement>;
+  title: string;
+  isTodoAdding?: boolean;
+  onSubmitForm: (event?: React.FormEvent<HTMLFormElement>) => void;
+  onInputChange: React.Dispatch<React.SetStateAction<string>>;
+  cancelEditing?: () => void;
+};
+
+export const NewTodoField: React.FC<Props> = memo((props) => {
+  const {
+    newTodoField,
+    title,
+    isTodoAdding,
+    onSubmitForm,
+    onInputChange,
+    cancelEditing,
+  } = props;
+
+  useEffect(() => {
+    // focus the element with `ref={newTodoField}`
+    if (newTodoField) {
+      if (newTodoField.current) {
+        newTodoField.current.focus();
+      }
+    }
+  }, []);
+
+  const handleEscPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (cancelEditing && event.key === 'Escape') {
+      cancelEditing();
+    }
+  };
+
+  return (
+    <form onSubmit={onSubmitForm}>
+      <input
+        data-cy="NewTodoField"
+        type="text"
+        ref={newTodoField}
+        className="todoapp__new-todo"
+        placeholder="What needs to be done?"
+        value={title}
+        onChange={(event) => onInputChange(event.currentTarget.value)}
+        onBlur={() => onSubmitForm()}
+        disabled={isTodoAdding}
+        onKeyDown={handleEscPress}
+        tabIndex={0}
+      />
+    </form>
```

Comment:
I honestly don't understand why, but your input looks a little bit different from the example 
![image](https://user-images.githubusercontent.com/102512764/216365089-275c8475-45d5-4293-8e45-6cf060dc3024.png)
![image](https://user-images.githubusercontent.com/102512764/216365310-983b566f-1324-4a2f-b861-d7259537e940.png)

If you know why, please fix this

Reply:
fixed